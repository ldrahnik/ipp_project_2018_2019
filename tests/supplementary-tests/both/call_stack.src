.IPPcode19
DEFVAR GF@a
DEFVAR GF@while
MOVE GF@while bool@false
CALL skip
MOVE GF@while bool@true
LABEL skip
WRITE string@\032&lt;not-tag/&gt;\032
JUMPIFEQ end GF@while bool@true
RETURN
LABEL end
