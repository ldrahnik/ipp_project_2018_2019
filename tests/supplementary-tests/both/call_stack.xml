<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode19">
 <instruction order="1" opcode="DEFVAR">
  <arg1 type="var">GF@a</arg1>
 </instruction>
 <instruction order="2" opcode="DEFVAR">
  <arg1 type="var">GF@while</arg1>
 </instruction>
 <instruction order="3" opcode="MOVE">
  <arg1 type="var">GF@while</arg1>
  <arg2 type="bool">false</arg2>
 </instruction>
 <instruction order="4" opcode="CALL">
  <arg1 type="label">skip</arg1>
 </instruction>
 <instruction order="5" opcode="MOVE">
  <arg1 type="var">GF@while</arg1>
  <arg2 type="bool">true</arg2>
 </instruction>
 <instruction order="6" opcode="LABEL">
  <arg1 type="label">skip</arg1>
 </instruction>
 <instruction order="7" opcode="WRITE">
  <arg1 type="string">\032&amp;lt;not-tag/&amp;gt;\032</arg1>
 </instruction>
 <instruction order="8" opcode="JUMPIFEQ">
  <arg1 type="label">end</arg1>
  <arg2 type="var">GF@while</arg2>
  <arg3 type="bool">true</arg3>
 </instruction>
 <instruction order="9" opcode="RETURN"/>
 <instruction order="10" opcode="LABEL">
  <arg1 type="label">end</arg1>
 </instruction>
</program>  
