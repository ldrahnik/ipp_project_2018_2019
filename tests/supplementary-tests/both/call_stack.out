 &lt;not-tag/&gt;  &lt;not-tag/&gt; 